src/main/java/net/neoforged/neoforge/common/extensions/IPacketFlowExtension.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.extensions;
+
+import net.minecraft.network.protocol.PacketFlow;
+import net.neoforged.fml.LogicalSide;
+
+/**
+ * Extension for {@link PacketFlow} to add some utility methods.
+ */
+public interface IPacketFlowExtension {
+
```

Comment:
This is an useless extension for enum checks.

Reply:
I would argue that this is a useful extension, especially when people work with streams or functional programming.

Reply:
`Predicate.isEqual` exists and does not require patches

Reply:
After internal discussion, there seems to be little consensus within the maintainer team; as the PR author, I decided to keep this in, as it matches my programming style better.

The isXXXX methods do not expand the patch size or the maintenance overhead.

Reply:
I would like to note my protest over this change, as I feel the principle of keeping an API small takes precedence over following the personal programming preferences of a maintainer-author and adding unnecessary methods to our public API surface.

There is no functional difference between `isClientbound()` and the equivalent check `== PacketFlow.CLIENTBOUND` (with the same for `SERVERBOUND`), and functional uses can be replaced with `Predicate.isEqual(PacketFlow.CLIENTBOUND)`.

The only substantial precedents for these kinds of methods exist in classes _outside_ of NeoForge: the `Dist` class (provided by [the MergeTool API](https://github.com/neoforged/MergeTool/blob/main/src/forgeAPI/java/net/neoforged/api/distmarker/Dist.java)) and the `LogicalSide` class (provided by [FancyModLoader](https://github.com/neoforged/FancyModLoader/blob/29b9b43185269582e8596ab68ecba362f988ef04/core/src/main/java/net/neoforged/fml/LogicalSide.java#L41-L55)).[^1] In both cases, these methods exist moreso by way of having existed long before many of us started developing NeoForge (or its predecessor, Minecraft Forge A.K.A. LexForge).

I would extremely hesitate to make precedent the pattern of including such 'helpers' in our API surface, particular for _very simple_ enums whose values are unlikely to ever change and where straight `==` comparisons are ideal. Insofar as this pattern goes, I will make it a point to bring up this discussion anytime this pattern appears in any contribution (by any author) in the future.

[^1]: Outside of these two cases, the only other uses of this design pattern -- having equivalent `boolean is<Value>()` methods for enum values which do `this == <VALUE>` -- are in NeoForge's `ConnectionType` and `IFluidHandler.FluidAction`; in the former case, only `isVanilla()` exists without an `isModded()` counterpart, and in the latter case, the `is` prefix is not present for `execute()` and `simulate()`.